app/views/result/index.html.erb

```diff
@@ -1,126 +1,140 @@
-  <% url += term.split.join("+") %>
+      <h2 class="head-text">New Search</h2>
 
+      <form id="homepage-form" action="/result" method="get">
 
-  <%# Allows the user to not enter a result count (default's 10)
-      and caps the displayed results at 1000 as per API limit.
-      Also adds the count to the URL that's passed in        %>
-  <% if params["count"] == "" %>
-    <% count = "10" %>
-  <% elsif params["count"].to_i > 1000 %>
-    <%count = "1000" %>
-  <% else %>
-    <%count = params["count"] %>
-  <% end %>
-  <% url += "&count=" + count %>
+        <div class="form-group">
+          <label class="form-label" for="search_term_input">
+            Search term to compare
+          </label>
+          <input class="form-control" id="search_term_input" type="text" name="search_term"></input>
+        </div>
 
+        <div class="form-group">
+          <label class="form-label" for="result_count">
+            Required number of results
+          </label>
+          <input class="form-control" id="result_count" name="count" type="number"></input>
+        </div>
 
-  <%# Creates an empty hash to put the search results from our 'A'
-      test in and a counter to increment within each of the following
-      loops                           %>
-  <% a_list = {} %>
-  <% a_counter = 0 %>
-  <% b_counter = 0 %>
+        <div class="form-group" id="hidden-search-group">
+          <label class="form-label" for="AB_select">
+            A/B test being used
+          </label>
+          <select class="form-control" name="which_test" id="AB_select">
+            <option value="search_match_length">Search Match Length</option>
+          </select>
+        </div>
 
-  <div class="AB_search_wrapper">
-    <h2>A Results</h2>
+        <input class="button" id="search_button" type="submit" value="Search"></input>
 
-    <%# Adds the A test parameter to the URL  %>
-    <% uri = URI("#{url}&ab_tests=#{params["which_test"]}:A") %>
+      </form>
 
-    <%# Gets a result from the search API based on the URL we made %>
-    <% response = Net::HTTP.get(uri) %>
-    <% findings = JSON.parse(response) %>
+    </div>
 
-    <%# Loops over the 'results' section of our result %>
-    <% findings["results"].each do |each_bit| %>
 
-      <%# increments counter and creates hash entry based on link and counter %>
-      <% a_counter += 1 %>
-      <% a_list["#{each_bit["link"]}"] = a_counter %>
 
-      <%# displays this in the 'A' section %>
-      <div class="changeless_box">
-        <% link = "https://gov.uk" + each_bit["link"] %>
-        <a href = "<%= link %>" ><%= each_bit["title"] %></a>
-      </div>
-      <br><br>
+    <%# Sets the base bit of the url and adds the user
+        defined search term connected with '+' so it's in
+        the right format                                  %>
+    <% url = 'https://www.gov.uk/api/search.json?q=' %>
+    <% term = params["search_term"] %>
+    <% url += term.split.join("+") %>
```

Comment:
You might have trouble with some special characters like & and ? here.

In general, we have to be careful when mixing strings that have special structure to them, because in some cases the user can craft particular inputs that make the app do something it shouldn't (cross site scripting is an example of this).

In this case you can avoid problems by url encoding: https://developer.mozilla.org/en-US/docs/Glossary/percent-encoding